src/app/page/Request/foodbox/FoodboxStep.jsx

```diff
@@ -1,64 +1,94 @@
 import Select from "@material-ui/core/Select";
-import _ from "lodash";
-import PropTypes from "prop-types";
-import React from "react";
+import React, { useEffect } from "react";
 import { useHistory } from "react-router-dom";
+import { Typography } from "@material-ui/core";
 
 import { ReactComponent as HappyFace } from "../../../../img/happy-face.svg";
 import { Body1 } from "../../../component";
-import { HappyBox, HR, StyledHeader, TotalsContainer, useStyles } from "./foodboxSteps.style";
+import { HappyBox, HR, TotalsContainer, useStyles } from "./foodboxSteps.style";
 import NavigationButtons from "./NavigationButtons";
+import Organization from "../../../model/Organization";
 
-function FoodboxStep({ handleChange, mockData, onNext, values }) {
+function FoodboxStep({ dispatch, state }) {
   const history = useHistory();
   const classes = useStyles();
+  const { cart } = state;
+
+  useEffect(() => {
+    // We only need to do this once to get the initial foodboxes
+    if (Object.keys(cart).length < 1) {
+      // only using first box for now
+      Organization.getFoodBoxes().then((boxes) =>
+        dispatch({ type: "UPDATE_CART", payload: { resource: boxes[0], quantity: 1 } })
+      );
+    }
+  }, [cart, dispatch]);
 
   const getTotalPrice = () => {
-    return (mockData.BASKET_PRICE * values.quantity).toFixed(2);
+    return Object.keys(cart).reduce(
+      (total, key) => cart[key].resource.cost * cart[key].quantity + total,
+      0
+    );
   };
 
   const getTotalBoxes = () => {
-    return `${values.quantity} box${values.quantity === 1 ? "" : "es"}`;
+    const total = Object.keys(cart).reduce((total, key) => cart[key].quantity + total, 0);
+    return `${total} box${total === 1 ? "" : "es"}`;
```

Comment:
Do we want to just put `box(es)` and skip the need for a calculation in the language?

Reply:
Guess it's more of a design question. figma has `box`